File: pset6/cash/cash.py
from cs50 import get_float


def get_change():
    change = -1

    while change < 0:
        change = get_float("Change: ")

    return int(change * 100)


change = get_change()
coins = 0

while change > 0:
    if change >= 25:
        change -= 25
        coins += 1
    elif change >= 10:
        change -= 10
        coins += 1
    elif change >= 5:
        change -= 5
        coins += 1
    else:
        change -= 1
        coins += 1

print(coins)

File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float get_owed(string p);

int main(void)
{
    float dollars = get_owed("Cash owed: ");
    int cents = round(dollars * 100);
    int count = 0;
    
    while (cents > 0)
    {
        // Check which is the biggest possible denomination that can be used and subtract it's vale
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else 
        {
            cents -= 1;
        }
        count++;
    }
    printf("%i\n", count);
}

// Take only possitive inputs
float get_owed(string p)
{
    float n;
    do
    {
        n = get_float("%s", p);
    } 
    while(n < 0);
    return n;
}
